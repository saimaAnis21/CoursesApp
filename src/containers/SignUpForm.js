import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';
import { connect } from 'react-redux';
import fetchAuthToken from '../logic/fetchAuthToken';
import { SaveAuthToken, SaveEmail } from '../logic/localStorage';
import { changeLogInfoAction } from '../actions/index';
import '../styles.css';

const SignUpForm = (props) => {
  const [name, setName] = useState();
  const [email, setEmail] = useState();
  const [password, setPassword] = useState();
  const [pwd, setPwd] = useState();
  const [msgs, setMsgs] = useState();
  const history = useHistory();
  const { logInfo } = props;

  const inputChange = (e) => {
    if (e.target.name === 'name') {
      setName(e.target.value);
    } else if (e.target.name === 'email') {
      setEmail(e.target.value);
    } else if (e.target.name === 'password') {
      setPassword(e.target.value);
    } else if (e.target.name === 'pwd') {
      setPwd(e.target.value);
    }
  };

  const getData = async () => {
    const queries = {
      name,
      email,
      password,
      pwd,
    };
    const response = await fetchAuthToken(queries, 'signup');
    if (response.auth_token) {
      SaveAuthToken(response.auth_token);
      SaveEmail(email);
      history.push('/my_courses');
      logInfo({
        logged_in: true,
        email,
      });
    } else if (response.message) {
      setMsgs(`${response.message}!!`);
    }
  };

  const btnClick = (e) => {
    e.preventDefault();
    getData();
  };

  return (
    <div className=" p-20 txt-align mx-auto mt-50 ">
      <p className="text-danger">
        {msgs}
      </p>
      <form className="">
        <div className="mt-10 mb-10 d-flex flex-column align-items-center mx-auto ">

          <div className="input-group mb-3">
            <div className="input-group-prepend">
              <span className="input-group-text">User Name</span>
            </div>
            <input type="text" onChange={(e) => inputChange(e)} id="name" name="name" className="form-control" />
          </div>

          <div className="input-group mb-3">
            <div className="input-group-prepend">
              <span className="input-group-text">Email</span>
            </div>
            <input type="text" onChange={(e) => inputChange(e)} id="email" name="email" className="form-control" />
          </div>

          <div className="input-group mb-3">
            <div className="input-group-prepend">
              <span className="input-group-text">Password</span>
            </div>
            <input type="password" onChange={(e) => inputChange(e)} id="password" name="password" className="form-control" />
          </div>

          <div className="input-group mb-3">
            <div className="input-group-prepend">
              <span className="input-group-text">Confirm Password</span>
            </div>
            <input type="password" onChange={(e) => inputChange(e)} id="pwd" name="pwd" className="form-control" />
          </div>

          <button
            type="submit"
            className="btn btn-dark btn-lg btn-block"
            onClick={(e) => btnClick(e)}
          >
            Sign Up
          </button>
        </div>

      </form>
    </div>
  );
};

const mapDispatchToProps = (dispatch) => ({
  logInfo: (info) => {
    dispatch(changeLogInfoAction(info));
  },
});

SignUpForm.propTypes = {
  logInfo: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(SignUpForm);
