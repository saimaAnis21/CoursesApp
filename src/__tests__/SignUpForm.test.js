import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import SignUpForm from '../containers/SignUpForm';

const store = createStore(() => [], {});
const Wrapper = ({ children }) => (
  <Provider store={store}>{children}</Provider>
);

Wrapper.propTypes = {
  children: PropTypes.shape({}).isRequired,
};

describe('Sign Up Form', () => {
  it('renders correctly', () => {
    const tree = render(<SignUpForm />, { wrapper: Wrapper });
    expect(tree).toMatchSnapshot();
  });
});
