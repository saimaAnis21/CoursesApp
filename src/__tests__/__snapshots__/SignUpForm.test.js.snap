// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Up Form renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" p-20 txt-align mx-auto mt-50 "
      >
        <p
          class="text-danger"
        />
        <p>
          Already have an account!
           
          <a
            href="/signin"
          >
            Sign in!
          </a>
        </p>
        <form
          class=""
        >
          <div
            class="mt-10 mb-10 d-flex flex-column align-items-center mx-auto "
          >
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                >
                  User Name
                </span>
              </div>
              <input
                class="form-control"
                id="name"
                name="name"
                type="text"
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                >
                  Email
                </span>
              </div>
              <input
                class="form-control"
                id="email"
                name="email"
                type="text"
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                >
                  Password
                </span>
              </div>
              <input
                class="form-control"
                id="password"
                name="password"
                type="text"
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                >
                  Confirm Password
                </span>
              </div>
              <input
                class="form-control"
                id="pwd"
                name="pwd"
                type="text"
              />
            </div>
            <button
              class="btn btn-dark btn-lg btn-block"
              type="submit"
            >
              Sign Up
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" p-20 txt-align mx-auto mt-50 "
    >
      <p
        class="text-danger"
      />
      <p>
        Already have an account!
         
        <a
          href="/signin"
        >
          Sign in!
        </a>
      </p>
      <form
        class=""
      >
        <div
          class="mt-10 mb-10 d-flex flex-column align-items-center mx-auto "
        >
          <div
            class="input-group mb-3"
          >
            <div
              class="input-group-prepend"
            >
              <span
                class="input-group-text"
              >
                User Name
              </span>
            </div>
            <input
              class="form-control"
              id="name"
              name="name"
              type="text"
            />
          </div>
          <div
            class="input-group mb-3"
          >
            <div
              class="input-group-prepend"
            >
              <span
                class="input-group-text"
              >
                Email
              </span>
            </div>
            <input
              class="form-control"
              id="email"
              name="email"
              type="text"
            />
          </div>
          <div
            class="input-group mb-3"
          >
            <div
              class="input-group-prepend"
            >
              <span
                class="input-group-text"
              >
                Password
              </span>
            </div>
            <input
              class="form-control"
              id="password"
              name="password"
              type="text"
            />
          </div>
          <div
            class="input-group mb-3"
          >
            <div
              class="input-group-prepend"
            >
              <span
                class="input-group-text"
              >
                Confirm Password
              </span>
            </div>
            <input
              class="form-control"
              id="pwd"
              name="pwd"
              type="text"
            />
          </div>
          <button
            class="btn btn-dark btn-lg btn-block"
            type="submit"
          >
            Sign Up
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
